File: MERLIN/MERLIN_PRSA_publish/Ogden.m
function [Sx, Et, Wb] = Ogden(Ex, C0)
% Ogden hyperelastic constitutive model for bar elements
alfa = [5,1]; % Specify parameteres
pstr = real(sqrt(2*Ex+1));
C0 = (pstr<1)*1*C0+(~(pstr<1))*1*C0;
Et = C0/(alfa(1)-alfa(2)).*((alfa(1)-2)*pstr.^(alfa(1)-4)-(alfa(2)-2)*pstr.^(alfa(2)-4));
Sx = C0/(alfa(1)-alfa(2)).*(pstr.^(alfa(1)-2)-pstr.^(alfa(2)-2));
if nargout>2
    Wb = C0/(alfa(1)-alfa(2)).*((pstr.^alfa(1)-1)/alfa(1)-(pstr.^alfa(2)-1)/alfa(2));
end
    

File: python/Ogden.py
import numpy as np
from scipy.sparse import issparse

def Ogden(Ex, C0):
    '''The function `Ogden` implements the Ogden hyperelastic constitutive model for bar elements and
    calculates stress, elastic modulus, and strain energy density based on specified parameters and
    input values.
    
    Parameters
    ----------
    Ex
        It seems like the description of the parameter Ex is missing. Could you please provide more
    information or context about what Ex represents in the context of the Ogden hyperelastic
    constitutive model for bar elements?
    C0
        C0 is a parameter in the Ogden hyperelastic constitutive model for bar elements. It represents the
    initial stiffness of the material.
    
    Returns
    -------
        The function `Ogden` returns three values: `Sx`, `Et`, and `Wb`.
    
    '''
    # Ogden hyperelastic constitutive model for bar elements
    alfa = [5, 1]  # Specify parameters  
    pstr = np.real(np.sqrt(2*Ex+1))
    C0 = (pstr<1)*1*C0+(~(pstr<1))*1*C0
    Et = C0/(alfa[0]-alfa[1])*((alfa[0]-2)*pstr**(alfa[0]-4)-(alfa[1]-2)*pstr**(alfa[1]-4))
    Sx = C0/(alfa[0]-alfa[1])*(pstr**(alfa[0]-2)-pstr**(alfa[1]-2))
    Wb = C0/(alfa[0]-alfa[1])*((pstr**alfa[0]-1)/alfa[0]-(pstr**alfa[1]-1)/alfa[1])
    return Sx, Et, Wb


